### ISSUE TITLE
Report rules whose paths match no files

### ISSUE THREAD

**Issue by asamarts** [asamarts/alint#257]
## Why

alint.org's `.alint.yml` pointed three prose rules at `public/llms.txt` from 2026-06-09, when that file moved to `src/data/llms.txt`, until 2026-09-24. For those three and a half months `alint check` reported that every rule passed, because a rule whose `paths` match nothing passes. It was found by a manual audit, not by alint.

Repro with 0.16.1:

```yaml
version: 1
rules:
  - id: vacuous
    kind: file_content_forbidden
    paths: "public/llms.txt"
    pattern: 'x'
    level: error
```

```
$ alint check
✓ All 1 rule(s) passed.
```

alint.org now pins the path with a `file_exists` rule, but every config has the same blind spot.

## Proposal

One or more of:

- A per-rule `expect_matches: true` that turns an empty scope into a violation. `manifest_path_set` already has `expect_nonempty` for the same idea.
- A run-level option (a flag or a config key) that reports every rule whose scope matched no files. Apply it after `when:` gating, so rules that are switched off aren't reported.
- Show the count of rules that matched nothing in the summary and in `--format json`, so it's visible without failing the run.

For content rules, a literal path (no glob characters) that matches nothing is almost always a mistake, so that case could warn by default. Existence rules (`file_exists`, `file_absent` and similar) target files that may be missing on purpose, so they are exempt.
